qiskit/providers/ibmq/ibmqbackend.py

```diff
@@ -226,8 +243,16 @@ def retrieve_job(self, job_id):
                                        .format(job_id, self.name()))
 
             # Check for pre-qobj jobs.
+            kwargs = {}
             try:
-                ApiJobKind(job_info.get('kind', None))
+                job_kind = ApiJobKind(job_info.get('kind', None))
+
+                if isinstance(self._api, IBMQClient):
```

Comment:
Non-blocking. Again, I would change this to:
```python
is_new_api = isinstance(self._api, IBMQClient)
requires_qobj_storage = job_kind is ApiJobKind.QOBJECT_STORAGE
```

And then pass both options explicitely at [line 269](https://github.com/Qiskit/qiskit-ibmq-provider/pull/144/files#diff-e2b490e8ae3a5cdb85265b098bea5665R269).